pkg/securitypolicy/securitypolicyenforcer.go

```diff
@@ -447,6 +470,77 @@ func possibleIndexesForID(containerID string, mapping map[int]map[string]struct{
 	return possibles
 }
 
+// EnforceExpectedMountsPolicy for StandardSecurityPolicyEnforcer injects a
+// hooks.CreateRuntime hook into container spec and the hook ensures that
+// the expected mounts appear prior container start. At the moment enforcement
+// is expected to take place inside LCOW UVM.
+func (pe *StandardSecurityPolicyEnforcer) EnforceExpectedMountsPolicy(containerID string, spec *oci.Spec) error {
+	pe.mutex.Lock()
+	defer pe.mutex.Unlock()
+
+	if len(pe.Containers) < 1 {
+		return errors.New("policy doesn't allow mounting containers")
+	}
+
+	sandboxID := spec.Annotations[annotations.KubernetesSandboxID]
+	if sandboxID == "" {
+		return errors.New("no sandbox ID present in spec annotations")
+	}
+
+	var wMounts []string
+	pIndices := possibleIndexesForID(containerID, pe.ContainerIndexToContainerIds)
+	if len(pIndices) == 0 {
+		return errors.New("no valid container indices found")
+	}
+
+	matchFound := false
+	for _, index := range pIndices {
+		if !matchFound {
+			matchFound = true
+			wMounts = pe.Containers[index].ExpectedMounts
+		} else {
+			pe.narrowMatchesForContainerIndex(index, containerID)
+		}
+	}
```

Comment:
Can you explain what is happening here? Wouldn't this just be the equivalent of running `narrorMatchesForContainerIndex` for all entries other than the first?

Reply:
since a single overlay can be used by multiple containers, we keep a mapping between a given overlay and potential container IDs the overlay can correspond to. In other places, the narrowing is done via checking command used to start the container or matching environment variables against what's in the rules.
In the above scenarios, the narrowing can be and is done for optimization purposes. Here, however, as you pointed out, there's no need to match against anything, since the expected mounts don't actually "force" any constraints other than adding hooks when needed. I just left the code pattern similar to other places, in case we actually need to stick some validation there in the future.

Reply:
Hmm okay, maybe a comment would be good here to say that.

Reply:
I added one before the loop